server/data/nomisClientBuilder.js

```diff
@@ -146,12 +152,68 @@ function nomisGetBuilder(tokenStore, signInService, username) {
     }
 }
 
-function addReleaseDate(prisoner) {
+function effectiveConditionalReleaseDate({conditionalReleaseDate, conditionalReleaseOverrideDate}) {
+    if (conditionalReleaseOverrideDate && conditionalReleaseOverrideDate !== invalidDate) {
+        return conditionalReleaseOverrideDate;
+    }
+
+    if (conditionalReleaseDate && conditionalReleaseDate !== invalidDate) {
+        return conditionalReleaseDate;
+    }
+
+    return null;
+}
+
+function effectiveAutomaticReleaseDate({automaticReleaseDate, automaticReleaseOverrideDate}) {
+    if (automaticReleaseOverrideDate && automaticReleaseOverrideDate !== invalidDate) {
+        return automaticReleaseOverrideDate;
+    }
+
+    if (automaticReleaseDate && automaticReleaseDate !== invalidDate) {
+        return automaticReleaseDate;
+    }
+
+    return null;
+}
```

Comment:
how about something like overrideIfPresent(standard, override){ etc }

Reply:
I like that idea.. will change